test/integration/policy_install_operator_test.go

```diff
@@ -0,0 +1,389 @@
+				Eventually(
+					common.GetComplianceState(policyNamePrefix+noGroupSuffix),
+					defaultTimeoutSeconds*2,
+					1,
+				).Should(Equal(policiesv1.NonCompliant))
+
+				By("Checking if the correct condition is generated")
+				Eventually(
+					common.GetOpPolicyCompMsg("operator-policy"+noGroupSuffix),
+					defaultTimeoutSeconds*2,
+					1,
+				).Should(MatchRegexp("NonCompliant.*the OperatorGroup required by the policy was not found.*" +
+					"the Subscription required by the policy was not found.*"))
+			})
+
+			It("Should enforce the policy on the hub", func() {
+				common.EnforcePolicy(policyNamePrefix + noGroupSuffix)
+				By("Checking if the status of the root policy is compliant")
+				Eventually(
+					common.GetComplianceState(policyNamePrefix+noGroupSuffix),
+					defaultTimeoutSeconds*2,
+					1,
+				).Should(Equal(policiesv1.Compliant))
+
+				Eventually(
+					common.GetOpPolicyCompMsg("operator-policy"+noGroupSuffix),
+					defaultTimeoutSeconds*2,
+					1,
+				).Should(MatchRegexp("Compliant.*the OperatorGroup matches what is required by the policy.*" +
+					"the Subscription matches what is required by the policy.*"))
+			})
+
+			It("Should verify OperatorGroup details", func() {
+				By("Getting the OperatorGroup name from relatedObj field")
+				opPolicy := utils.GetWithTimeout(
+					clientManagedDynamic,
+					common.GvrOperatorPolicy,
+					"operator-policy"+noGroupSuffix,
+					clusterNamespace,
+					true,
+					defaultTimeoutSeconds,
+				)
+				Expect(opPolicy).NotTo(BeNil())
+
+				relObjList, found, err := unstructured.NestedSlice(opPolicy.Object, "status", "relatedObjects")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).Should(BeTrue())
+
+				foundOpGroupName := ""
+				for _, relObj := range relObjList {
+					relObjMap, ok := relObj.(map[string]interface{})
+					if !ok {
+						continue
+					}
+
+					kind, _, _ := unstructured.NestedString(relObjMap, "object", "kind")
+
+					if kind == "OperatorGroup" {
+						foundOpGroupName, _, _ = unstructured.NestedString(relObjMap, "object", "metadata", "name")
+					}
+				}
+				Expect(foundOpGroupName).ToNot(BeEmpty())
+
+				dynamicOpGroupName = foundOpGroupName
+
+				opGroup := utils.GetWithTimeout(
+					clientManagedDynamic,
+					common.GvrOperatorGroup,
+					foundOpGroupName,
+					testNS+noGroupSuffix,
+					true,
+					defaultTimeoutSeconds,
+				)
+				Expect(opGroup).ToNot(BeNil())
+			})
+
+			It("Should verify Subscription details", func() {
+				sub := utils.GetWithTimeout(
+					clientManagedDynamic,
+					common.GvrSubscriptionOLM,
+					subName,
+					testNS+noGroupSuffix,
+					true,
+					defaultTimeoutSeconds,
+				)
+				Expect(sub).NotTo(BeNil())
+			})
+
+			It("Should verify CSV details", func() {
+				csv := utils.GetWithTimeout(
+					clientManagedDynamic,
+					common.GvrClusterServiceVersion,
+					csvName,
+					testNS+noGroupSuffix,
+					true,
+					defaultTimeoutSeconds*4,
+				)
+				Expect(csv).NotTo(BeNil())
```

Comment:
Are there any details in the CSV we want to check? Maybe this whole thing could be in an Eventually waiting for a certain status?